openedx/core/djangoapps/programs/signals.py

```diff
@@ -51,4 +52,4 @@ class of the object instance that sent this signal
     )
     # import here, because signal is registered at startup, but items in tasks are not yet able to be loaded
     from openedx.core.djangoapps.programs.tasks.v1.tasks import award_program_certificates
-    award_program_certificates.delay(user.username)
+    award_program_certificates.delay(user.username, get_current_site().id)
```

Comment:
We caught up about this offline, but for the record: this receiver isn't always run in a process that has access to a request. For example, consider what happens when course staff decide to issue certificates manually. Without access to a request, `get_current_site()` won't bring back the correct site.

Another way of achieving what you're after here is to only pass the username argument to the task like before, then loop over each site within the task code when looking for programs that the user has completed.